keras_cv/visualization/segmentation_mask.py

```diff
@@ -0,0 +1,146 @@
+
+
+def _map_color_on_mask(masks, color):
+
+    # check distinct mask color codes with color mapping.
+    distinct_mask_code = tf.unique(tf.reshape(masks, -1)).y
+    color_rgb = defaultdict(tuple)
+
+    if isinstance(color, str):
+        for idx in distinct_mask_code:
+            try:
+                color_rgb.update({int(idx): colors[color]})
+            except KeyError:
+                raise KeyError(
+                    f"{color} is not supported yet,"
+                    "please check supported colors at `keras_cv.visualization.colors`"
+                )
+
+        color_keys = list(distinct_mask_code.numpy())
+
+    else:
+        if any([code not in distinct_mask_code for code in color.keys()]):
+            raise ValueError(
+                f"Color mapping {color} does not map completely\
+                  with distint color codes present in masks: {distinct_mask_code}"
+            )
+
+        # map color code with RGB
+        for code, name in color.items():
+            try:
+                color_rgb.update({code: colors[name]})
+            except KeyError:
+                raise KeyError(
+                    f"{name} is not supported yet,"
+                    "please check supported colors at `keras_cv.visualization.colors`"
+                )
+        color_keys = list(color.keys())
+
+    keys_tensor = tf.constant(color_keys, dtype=tf.int32)
+    colored_masks = tf.TensorArray(
+        tf.int32, size=0, dynamic_size=True, clear_after_read=True
+    )
+
+    for c in range(RGB):
+        vals_tensor = tf.constant([color_rgb[color][c] for color in color_keys])
+        table = tf.lookup.StaticHashTable(
+            tf.lookup.KeyValueTensorInitializer(keys_tensor, vals_tensor),
+            default_value=0,
+        )
+        colored_masks.write(c, table[tf.cast(masks, tf.int32)]).mark_used()
+
+    return tf.transpose(colored_masks.stack(), list(range(1, masks.ndim + 1)) + [0])
+
+
+def draw_segmentation(image, mask, color={}, alpha=0.4):
+    """Draws segmentation masks on images with desired color
+    and transparency.
+    Colors supported are standard X11 colors.
+
+    Args:
+        image: An integer tensor with shape (N, img_height, img_height, 3)
+        mask: An integer tensor of shape (N, img_height, img_height) with masks
+              values corresponding to `color_map`.
+        color: The color or colors to draw the segmentation map/s in.
+               This can either be a single color, or a mapping from class IDs
+               to colors.  Supported color formats are RGB tuples or strings.
+               A full list of color strings is available in keras_cv/visualization/colors.py
+        alpha: Transparency value between 0 and 1. (default: 0.4)
+    Returns:
+        Masks overlaid on images.
+
+    Raises:
+        ValueError: On incorrect data type and shapes for images or masks.
+        KeyError: On incorrect color in `color_map`.
+    """
+    tf.debugging.assert_integer(
```

Comment:
We need to support float dtypes for images.

Reply:
okay, let me work on it

Reply:
done